docs/src/main/asciidoc/oauth2-guide.adoc

```diff
@@ -0,0 +1,357 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Using OAuth2 RBAC
+
+include::./attributes.adoc[]
+:extension-name: Elytron Security OAuth2
+:oauth2-token: OAuth2 tokens
+
+This guide explains how your Quarkus application can utilize {oauth2-token} to provide secured access to the JAX-RS endpoints.
+
+If your OAuth2 Authentication server provides JWT tokens, you should use link:jwt-guide.html[MicroProfile JWT RBAC] instead, this extension aims to be used with obscure tokens and validate the token by calling an introspection endpoint.
+
+== Configuration
+
```

Comment:
Maybe a small intro would be useful.

Reply:
Maybe something like this ?

```
OAuth 2 is an authorization framework that enables applications to obtain access to an HTTP resource on behalf of a user. 
It can be used to implement an application authentication mechanism based on tokens by delegating to an external server (the authentication server) the user authentification and providing a token for the authentication context.
```

In fact it's always complex to define what is OAuth2 because it's not an authentication protocole but it can be used to build one ...

Reply:
Your proposal looks fine to me except for `s/OAuth 2/OAuth2/`.

Reply:
done